### PR TITLE
Fix/refactor

### PR DESCRIPTION
**Pull request by nov1n**
Add comments to each function except for manager.go as this will be split up into multiple files

### CODE REVIEW

File: pkg/api/types.go [hunk 1]
```diff
@@ -22,69 +22,73 @@ import (
...
 	k8sApi.ObjectMeta `json:"metadata,omitempty"`
 
+	// Maximum time before the Workflow may run before it is killed
 	ActiveDeadlineSeconds *int64 `json:"activeDeadlineSeconds,omitempty"`
 
+	// Steps maps step names to WorkflowStep objects for O(1) access by name
 	Steps map[string]WorkflowStep `json:"steps,omitempty"`
 
-	// Selector for created jobs (if any)
+	// Selector for created jobs
```
Comment (borismattijssen) [inline pkg/api/types.go]: `// Selector for jobs belonging to workflow`?

File: pkg/client/cache/listers.go [hunk 2]
```diff
@@ -76,7 +77,8 @@ func (s *StoreToWorkflowLister) GetJobWorkflows(job *k8sBatch.Job) (workflows []
 	return
 }
 
-// StoreToJobLister gives a store List and Exists methods. The store must contain only Jobs.
+// StoreToJobLister gives a store List and Exists methods.
```
Comment (borismattijssen) [inline pkg/client/cache/listers.go]: 'provides a store with..'?

File: pkg/client/workflows.go [hunk 3]
```diff
@@ -68,14 +68,18 @@ func newWorkflows(c *ThirdPartyClient, namespace string) *workflows {
 	}
 }
 
+// Update updates the given workflow in the cluster
 func (w *workflows) Update(workflow *api.Workflow) (result *api.Workflow, err error) {
 	return w.UpdateWithSubresource(workflow, "")
 }
 
+// UpdateStatus updates the status field for a given workflow TODO: Is this correct?
```
Comment (borismattijssen) [inline pkg/client/workflows.go]: Yes it's correct, but we don't use it.

File: pkg/client/workflows.go [hunk 4]
```diff
@@ -120,6 +124,7 @@ func (w *workflows) UpdateWithSubresource(workflow *api.Workflow, subresource st
 	return
 }
 
+// List lists returns a WorkflowList containing all workflows in the workflows namespace
```
Comment (borismattijssen) [inline pkg/client/workflows.go]: omit 'lists'

File: pkg/controller/controller_utils.go [hunk 5]
```diff
@@ -33,21 +33,22 @@ import (
...
+// JobControlInterface defines methods for JobControl
 type JobControlInterface interface {
-	// CreateJob
 	CreateJob(namespace string, template *k8sBatch.JobTemplateSpec, object k8sRunt.Object, key string) error
-	// DeleteJob
 	DeleteJob(namespace, name string, object k8sRunt.Object) error
 }
 
-// RealJobControl is the default implementation of JobControlInterface
+// WorkflowJobControl is the default implementation of JobControlInterface
```
Comment (borismattijssen) [inline pkg/controller/controller_utils.go]: is the 'workflow' implementation

File: pkg/controller/controller_utils.go [hunk 6]
```diff
@@ -33,21 +33,22 @@ import (
...
 type WorkflowJobControl struct {
 	KubeClient k8sClSet.Interface
 	Recorder   k8sRec.EventRecorder
 }
 
+// TODO: What is this?
```
Comment (borismattijssen) [inline pkg/controller/controller_utils.go]: dunno

File: pkg/validation/validation.go [hunk 7]
```diff
@@ -213,6 +223,8 @@ func ValidateWorkflowSpecUpdate(spec, oldSpec *api.WorkflowSpec, running, comple
 	return allErrs
 }
 
+// ValidateWorkflowStatusUpdate validates a spec update checking if it should be allowed
```
Comment (borismattijssen) [inline pkg/validation/validation.go]: 'status' update

File: pkg/client/workflows.go (hunk 3 above)
Comment (nov1n) [inline pkg/client/workflows.go]: But it is identical to the function 'Update' right above it? shouldnt the UpdateWithSubresource call have something other than the empty string as second argument?

File: pkg/controller/controller_utils.go (hunk 6 above)
Comment (nov1n) [inline pkg/controller/controller_utils.go]: I removed the TODO but its a scary line

**Pull request merged**